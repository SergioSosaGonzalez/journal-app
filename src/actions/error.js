import { types } from '../types/types';

export const setError = (text) => ({ type: types.uiSetError, payload: text });

export const removeError = () => ({
  type: types.uiRemoveError,
});
