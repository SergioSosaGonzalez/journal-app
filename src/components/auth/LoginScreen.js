import React from 'react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { useForm } from '../../hooks/useForm';

import {
  startGoogleLogin,
  startLoginWithEmailPassword,
} from '../../actions/auth';

const LoginScreen = () => {
  const dispatch = useDispatch();
  const [formValues, handleInputChange] = useForm({
    email: 'sergioarsosa95@gmail.com',
    password: '123456789',
  });

  const { loading } = useSelector((state) => state.ui);

  const { email, password } = formValues;

  const handleLogin = (e) => {
    e.preventDefault();
    dispatch(startLoginWithEmailPassword(email, password));
  };

  const handleGoogleLogin = () => {
    dispatch(startGoogleLogin());
  };

  return (
    <>
      <h3 className="auth__title">Login</h3>
      <form
        onSubmit={handleLogin}
        className="animate__animated animate__fadeIn animate__faster">
        <input
          type="email"
          placeholder="email"
          className="auth__input"
          name="email"
          autoComplete="off"
          value={email}
          onChange={handleInputChange}
        />
        <input
          type="password"
          placeholder="Password"
          className="auth__input"
          name="password"
          value={password}
          onChange={handleInputChange}
        />

        <button
          disabled={loading}
          type="submit"
          className={`btn btn-primary btn-block ${loading ? 'disabled' : ''}`}>
          Login
        </button>
        <hr />
        <div className="auth__social-networks">
          <p>Login with Social Networks</p>
          <div className="google-btn" onClick={handleGoogleLogin}>
            <div className="google-icon-wrapper">
              <img
                className="google-icon"
                src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
                alt="google button"
              />
            </div>
            <p className="btn-text">
              <b>Sign in with google</b>
            </p>
          </div>
        </div>
        <Link to="/auth/register" className="link">
          Create new account
        </Link>
      </form>
    </>
  );
};
export default LoginScreen;
