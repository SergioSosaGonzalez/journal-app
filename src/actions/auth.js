import { types } from '../types/types';
import { firebase, googleAuthProvider } from '../firebase/firebase-config';
import Swal from 'sweetalert2';
import { notesLogout } from './notes';

export const startLoginWithEmailPassword = (email, password) => {
  return (dispatch) => {
    dispatch({ type: types.uiStartLoading, payload: {} });
    return firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(({ user }) => {
        dispatch({ type: types.uiFinishLoading, payload: {} });
        dispatch(login(user.uid, user.displayName));
      })
      .catch((e) => {
        Swal.fire('Error', e.message, 'error');
        dispatch({ type: types.uiFinishLoading, payload: {} });
      });
  };
};

export const startRegisterWithEmailPasswordName = (email, password, name) => {
  return (dispatch) => {
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(async ({ user }) => {
        await user.updateProfile({ displayName: name });
        dispatch(login(user.uid, user.displayName));
      })
      .catch((e) => {
        Swal.fire('Error', e.message, 'error');
      });
  };
};

export const startGoogleLogin = () => {
  return (dispatch) => {
    console.log('hola mudno');
    firebase
      .auth()
      .signInWithPopup(googleAuthProvider)
      .then(({ user }) => {
        dispatch(login(user.uid, user.displayName));
      });
  };
};

export const login = (uid, displayName) => {
  return {
    type: types.login,
    payload: {
      uid,
      displayName,
    },
  };
};

export const logout = () => {
  return { type: types.logout };
};

export const startLogout = () => {
  return async (dispatch) => {
    await firebase.auth().signOut();
    dispatch(logout());
    dispatch(notesLogout());
  };
};
