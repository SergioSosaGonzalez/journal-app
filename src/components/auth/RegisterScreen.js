import React from 'react';
import { Link } from 'react-router-dom';
import { useForm } from '../../hooks/useForm';
import validator from 'validator';
import { useDispatch, useSelector } from 'react-redux';
import { removeError, setError } from '../../actions/error';
import { startRegisterWithEmailPasswordName } from '../../actions/auth';

const RegisterScreen = () => {
  const dispatch = useDispatch();
  const { msgError } = useSelector((state) => state.ui);
  let [values, handleInputChange] = useForm({
    name: '',
    email: '',
    password: '',
    confirm: '',
  });
  const { name, email, password, confirm } = values;

  const handleRegister = (e) => {
    e.preventDefault();
    if (isFormValid()) {
      dispatch(startRegisterWithEmailPasswordName(email, password, name));
    } else console.log('no registrado');
  };

  const isFormValid = () => {
    if (name.trim().length === 0) {
      dispatch(setError('Falta campo nombre'));
      return false;
    } else if (email.trim().length === 0 || !validator.isEmail(email)) {
      dispatch(setError('Email no valido'));
      return false;
    } else if (password !== confirm || password.length < 5) {
      dispatch(setError('La contraseÃ±a debe ser menor a 5 caracteres'));
      return false;
    }
    dispatch(removeError());
    return true;
  };
  return (
    <>
      <h3 className="auth__title">Register</h3>
      <form
        onSubmit={handleRegister}
        className="animate__animated animate__fadeIn animate__faster">
        {msgError && <div className="auth__alert-error">{msgError}</div>}
        <input
          type="tex"
          placeholder="Name"
          className="auth__input"
          name="name"
          autoComplete="off"
          value={name}
          onChange={handleInputChange}
        />
        <input
          type="email"
          placeholder="email"
          className="auth__input"
          name="email"
          autoComplete="off"
          value={email}
          onChange={handleInputChange}
        />
        <input
          type="password"
          placeholder="Password"
          className="auth__input"
          name="password"
          value={password}
          onChange={handleInputChange}
        />
        <input
          type="password"
          placeholder="Confirm Password"
          className="auth__input"
          name="confirm"
          value={confirm}
          onChange={handleInputChange}
        />
        <button type="submit" className="btn btn-primary btn-block mb-5">
          Register
        </button>
        <Link to="/auth/login" className="link">
          Are you already register?
        </Link>
      </form>
    </>
  );
};
export default RegisterScreen;
