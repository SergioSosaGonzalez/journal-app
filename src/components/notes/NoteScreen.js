import React, { useEffect, useRef } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { activeNote, startDeleting } from '../../actions/notes';
import { useForm } from '../../hooks/useForm';
import { NoteAppbar } from './NoteAppbar';

const NoteScreen = () => {
  const dispatch = useDispatch();
  const { active: note } = useSelector((state) => state.notes);
  const [formValues, handleInputChange, reset] = useForm(note);
  const { body, title, url } = formValues;

  const activeId = useRef(note.id);

  useEffect(() => {
    if (note.id !== activeId.current) {
      reset(note);
      activeId.current = note.id;
    }
  }, [note, reset]);

  useEffect(() => {
    dispatch(activeNote(note.id, { ...formValues }));
  }, [formValues, dispatch]);

  const handleDelete = () => {
    dispatch(startDeleting(note.id));
  };
  return (
    <div className="notes__main-content">
      <NoteAppbar />
      <div className="notes__content">
        <form>
          <input
            type="text"
            placeholder="Some awesome title"
            className="notes__title-input"
            value={title}
            name="title"
            onChange={handleInputChange}
          />
          <textarea
            placeholder="What happen today"
            className="notes__text-area"
            value={body}
            name="body"
            onChange={handleInputChange}></textarea>
          <div className="notes__image">
            {note.url && <img src={note.url} alt="img" />}
          </div>
        </form>
      </div>

      <button className="btn btn-danger" onClick={handleDelete}>
        Delete
      </button>
    </div>
  );
};
export default NoteScreen;
