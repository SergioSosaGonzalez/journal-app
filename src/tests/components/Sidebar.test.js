import '@testing-library/jest-dom';
import React from 'react';
import Sidebar from '../../components/journal/Sidebar';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store'; //ES6 modules
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { startLogout } from '../../actions/auth';
import { startNewNote } from '../../actions/notes';

jest.mock('../../actions/auth', () => ({
  startLogout: jest.fn(),
}));

jest.mock('../../actions/notes', () => ({
  startNewNote: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
  auth: {
    uid: '1',
    name: 'Sergio',
  },
  ui: {
    loading: false,
    msgError: null,
  },
  notes: {
    active: null,
    notes: [],
  },
};

let store = mockStore(initState);
store.dispatch = jest.fn();

describe('Pruebas para el Sidebar', () => {
  const wrapper = mount(
    <Provider store={store}>
      <Sidebar />
    </Provider>
  );
  test('inicializacion del sidebar', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de llamar al logout', () => {
    const button = wrapper.find('.btn');
    button.simulate('click', {});
    expect(startLogout).toHaveBeenCalled();
  });

  test('Debe de llamar al startNewNote', () => {
    const button = wrapper.find('.journal__new-entry');
    button.simulate('click', {});
    expect(startNewNote).toHaveBeenCalled();
  });
});
