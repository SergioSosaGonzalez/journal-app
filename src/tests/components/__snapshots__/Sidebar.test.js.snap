// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el Sidebar inicializacion del sidebar 1`] = `
<aside
  className="journal__sidebar"
>
  <div
    className="journal__sidebar-navbar"
  >
    <h3
      className="mt-5"
    >
      <i
        className="far fa-moon"
      />
      <span>
         
        Sergio
      </span>
    </h3>
    <button
      className="btn"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
  <div
    className="journal__new-entry"
    onClick={[Function]}
  >
    <h3
      className="mt-5"
    >
      <i
        className="far fa-calendar-plus fa-5x"
      />
      <p
        className="mt-5"
      >
        New entry
      </p>
    </h3>
  </div>
  <div
    className="journal__entries "
  />
</aside>
`;
